hazelcast/src/main/java/com/hazelcast/internal/partition/impl/MigrationManager.java

```diff
@@ -1161,6 +1164,15 @@ public void run() {
             if (!partitionStateManager.isInitialized()) {
                 return;
             }
+            if (!node.getClusterService().getClusterState().isJoinAllowed()) {
```

Comment:
Is this check only needed in the `MigrationRunnable`s that modify the partition table? Is it needed in `MigrateTask`?

Reply:
Yes this is only needed tasks those modify partition table. `MigrateTask`s are produced by only `RepartitioningTask`.

Reply:
I've noticed that, not sure if we can have a `MigrateTask` in the migration queue after the state was changed. As I'm writing this down, I'm thinking not. :)

Reply:
We don't allow cluster state changes when there are tasks in the queue or a task is being executed.